controllers/provisioning_controller.go

```diff
@@ -333,8 +325,13 @@ func (r *ProvisioningReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	return result, nil
 }
 
-func (r *ProvisioningReconciler) provisioningInfo(provConfig *metal3iov1alpha1.Provisioning, images *provisioning.Images, sshkey string) (*provisioning.ProvisioningInfo, error) {
-	proxy, err := r.OSClient.ConfigV1().Proxies().Get(context.Background(), "cluster", metav1.GetOptions{})
+func (r *ProvisioningReconciler) provisioningInfo(ctx context.Context, provConfig *metal3iov1alpha1.Provisioning, images *provisioning.Images, sshkey string) (*provisioning.ProvisioningInfo, error) {
+	proxy, err := r.OSClient.ConfigV1().Proxies().Get(ctx, "cluster", metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	macAddresses, err := r.masterMacAddresses(ctx)
```

Comment:
Q: why are we even relying on the BootMACAddress given that many of the metal3 drivers (e.g. IPMI) don't even require it to be present?

Reply:
that was the basis for this epic: https://issues.redhat.com/browse/MPINSTALL-6?focusedCommentId=16032926&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16032926

Reply:
@zaneb FYI although provisioning can work for IPMI without bootMACAddress - adopt will then fail without it ref https://bugzilla.redhat.com/show_bug.cgi?id=1854142

@derekhiggins looked into changing that Ironic behavior ref https://bugzilla.redhat.com/show_bug.cgi?id=1876573 but AFAIK we've not found a way to remove that constraint, so the installer currently does treat bootMACAddress as mandatory in the install-config schema (hence, it seemed reasonable to leverage this existing data to identify the provisioning NIC)

Reply:
Perhaps it would be reasonable to only consume the install-config data (which is mandatory) to set this provisioning CR value, avoiding any confusion with the BMH field which is theoretically-optional-but-actually-required?

Reply:
Do we need to fix https://github.com/metal3-io/baremetal-operator/blob/master/pkg/bmc/ipmi.go#L60 then?
Do other examples (like https://github.com/metal3-io/baremetal-operator/blob/master/pkg/bmc/irmc.go#L36) have the same problem?

Reply:
@derekhiggins can you confirm - IIRC the adopt behavior always requires the MAC regardless of the driver (but on deploy some drivers work without a MAC because they can correlate a unique BMC address), but not completely sure I'm recalling the details correctly?

Reply:
@hardys ya there are some inconsistencies between drivers and when exactly the MAC is required, iirc pxe requires it on adopt but redfish-vmedia doesn't (I didn't look at any others) the safest option is to always specify it. I'm fuzzy on the exact details of when each driver requires it, I'd have to revisit this bug to see if it can be made more consistent.

Reply:
I opened https://github.com/metal3-io/baremetal-operator/issues/979 to track this.